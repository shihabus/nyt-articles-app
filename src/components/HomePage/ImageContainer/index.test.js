/* eslint-disable no-undef */
// /* eslint-disable import/named */
// /* eslint-disable no-undef */
// /**
//  * @jest-environment jsdom
//  */

import React from "react";
import { shallow } from "enzyme";
import { createSerializer } from "enzyme-to-json";
import Image from "./index";
import { checkProps,findComponent } from "../../../../testUtils";

expect.addSnapshotSerializer(createSerializer({ mode: "deep" }));

const setUp = (props = {}) => {
  const wrapper = shallow(<Image {...props} />);
  return wrapper;
};

describe("Image", () => {
  let wrapper;
  beforeEach(() => {
    const props = {
        urlToImage: 'Test'
    };
    wrapper = setUp(props);
  });

  it("Snapshot", () => {
    expect(wrapper).toMatchSnapshot("Image");
  });

  it("Checked props", () => {
    const expectedProps = {
        urlToImage: 'Test'
    };
    const propsErr = checkProps(Image, expectedProps);
    expect(propsErr).toBeUndefined();
  });

  it('should render child',()=>{
    expect(findComponent(wrapper,"Image").length).toEqual(1)
  })
});
