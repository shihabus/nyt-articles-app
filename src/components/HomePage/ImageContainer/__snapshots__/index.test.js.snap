// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Image Snapshot: Image 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
      "justifyContent": "center",
      "width": "20%",
    }
  }
>
  <Image
    resizeMode="contain"
    source={
      Object {
        "uri": "Test",
      }
    }
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "#CED0CE",
        "borderRadius": 25,
        "height": 50,
        "width": 50,
      }
    }
    testID="Image"
  />
</View>
`;
